### PR TITLE
Baseball!

### PR DESCRIPTION
**Pull request by Couls**
**What does this PR do:**
Baseball bats can now deflect normal and smaller sized objects, after a non-ball object is deflected there is a five minute cooldown use the baseball bat in hand to enter deflect mode, you cannot attack people during deflect mode

**Images of sprite/map changes (IF APPLICABLE):**
Baseball:
image Not yet actually used but could be put in a baseball themed holodeck or the like.

**Changelog:**
:cl:
add: You can now homerun things thrown at you with a baseball bat!
tweak: new baseball hit noise, new baseball sprite
/:cl:

### CODE REVIEW

**Comment by taukausanake** [comment]
Base _ball_ or baseball _bat_?

You know what, I don't even. You keep doing you

**Comment by Couls** [comment]
it's been a long day

**Comment by Ralta** [comment]
Awesome. Don't suppose it's possible to allow the baseball bat to always deflect baseballs? That way, you could use the bat every 5 mins to deflect a small/normal item, but could actually play baseball 'properly'.

Also, someone add a catchers mitt :stuck_out_tongue:

**Comment by Couls** [comment]
Yep! No cooldown for any balls including beach balls and basketballs!

File: code/modules/mob/living/carbon/human/human_defense.dm [hunk 1]
```diff
@@ -311,6 +311,39 @@ emp_act
...
+				visible_message("<span class='warning'><b>[src] Deflects the [I] directly back at the thrower! It's a home run!</b></span>", "<span class='warning'><b>You deflects the [I] directly back at the thrower!, It's a home run!</b></span>")
+				playsound(get_turf(src), 'sound/weapons/homerun.ogg', 100, 1)
+				do_attack_animation(I, ATTACK_EFFECT_DISARM)				
+				I.throw_at(I.thrownby, 20, 20, src)
+				B.deflectmode = FALSE
+				if(!istype(I, /obj/item/beach_ball))
+					B.lastdeflect = world.time + 3000
+				return TRUE				
+			else if(prob(30))
+				visible_message("<span class='warning'>[src] swings! And he misses! How embarassing.</span>", "<span class='warning'>You swing! You miss! Oh no!</span>")
```
Comment (Tayyyyyyy) [inline code/modules/mob/living/carbon/human/human_defense.dm]:
```suggestion
				visible_message("<span class='warning'>[src] swings! And [p_they()] miss[p_es()]! How embarassing.</span>", "<span class='warning'>You swing! You miss! Oh no!</span>")
```

File: code/game/objects/items/weapons/weaponry.dm [hunk 2]
```diff
@@ -171,12 +173,18 @@
 
 /obj/item/melee/baseball_bat/attack_self(mob/user)
 	if(!homerun_able)
-		..()
-		return
+		if(!deflectmode && world.time >= lastdeflect)
+			to_chat(user, "<span class='notice'>You prepare to deflect objects thrown at you, You cannot attack during this time.</span>")
```
Comment (Tayyyyyyy) [inline code/game/objects/items/weapons/weaponry.dm]: Comma should be a period here.

File: code/game/objects/items/weapons/weaponry.dm [hunk 3]
```diff
@@ -171,12 +173,18 @@
 
 /obj/item/melee/baseball_bat/attack_self(mob/user)
 	if(!homerun_able)
-		..()
-		return
+		if(!deflectmode && world.time >= lastdeflect)
+			to_chat(user, "<span class='notice'>You prepare to deflect objects thrown at you, You cannot attack during this time.</span>")
+			deflectmode = TRUE
+		else if(deflectmode && world.time >= lastdeflect)
+			to_chat(user, "<span class='notice'>You no longer deflect objects thrown at you, You can attack during this time</span>")
```
Comment (Tayyyyyyy) [inline code/game/objects/items/weapons/weaponry.dm]: Comma to period

File: code/modules/mob/living/carbon/human/human_defense.dm [hunk 4]
```diff
@@ -311,6 +311,39 @@ emp_act
...
 			return ..()
+	if(I.w_class <= WEIGHT_CLASS_NORMAL || istype(I, /obj/item/beach_ball)) // baseball bat deflecting
+		var/obj/item/melee/baseball_bat/B
+		if(l_hand && istype(l_hand, /obj/item/melee/baseball_bat))
+			B = l_hand
+		else if(r_hand && istype(r_hand, /obj/item/melee/baseball_bat))
+			B = r_hand
+		if(B && B.deflectmode)
+			if(prob(10))
+				visible_message("<span class='warning'><b>[src] Deflects the [I] directly back at the thrower! It's a home run!</b></span>", "<span class='warning'><b>You deflects the [I] directly back at the thrower!, It's a home run!</b></span>")
```
Comment (Tayyyyyyy) [inline code/modules/mob/living/carbon/human/human_defense.dm]:
```suggestion
				visible_message("<span class='warning'><b>[src] Deflects the [I] directly back at the thrower! It's a home run!</b></span>", "<span class='warning'><b>You deflect the [I] directly back at the thrower! It's a home run!</b></span>")
```
There's also a class called "boldwarning" that might be better

File: code/modules/mob/living/carbon/human/human_defense.dm [hunk 5]
```diff
@@ -311,6 +311,39 @@ emp_act
...
+			else if(prob(30))
+				visible_message("<span class='warning'>[src] swings! And he misses! How embarassing.</span>", "<span class='warning'>You swing! You miss! Oh no!</span>")
+				playsound(get_turf(src), 'sound/weapons/thudswoosh.ogg', 50, 1, -1)
+				do_attack_animation(get_step(src,pick(alldirs)), ATTACK_EFFECT_DISARM)			
+				B.deflectmode = FALSE
+				if(!istype(I, /obj/item/beach_ball))
+					B.lastdeflect = world.time + 3000
+				return ..()
+			else
+				visible_message("<span class='warning'>[src] swings and deflects the [I]!</span>", "<span class='warning'>You swing and deflect the [I]!</span>")
```
Comment (Tayyyyyyy) [inline code/modules/mob/living/carbon/human/human_defense.dm]: Isn't byond going to handle "the" insertion?

Comment (Couls) [reply code/modules/mob/living/carbon/human/human_defense.dm]: It really depends on how consistent our item naming is, but when tested it just gave the name of the item

**Comment by MRgaty** [comment]
F

**Comment by Fox-McCloud** [comment]
Snowflaking this into human defense is awful.

You should use `hit_reaction` for a baseball bat to do this sort of stuff.

**Comment by Couls** [comment]
Will look into it when I get the time, thanks for the tip

**Pull request merged**